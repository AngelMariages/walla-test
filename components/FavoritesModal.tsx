import { Item } from '../pages/api/items';
import Product from './Product';

type Props = {
    visible: boolean;
    onClose: () => void;
    favorites: Item[];
};

const FavoritesModal: React.FC<Props> = ({ visible, onClose, favorites }) => {
    if (!visible) return null;

    return (
        <div className="z-10 fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center">
            <div
                className="fixed inset-0 bg-black bg-opacity-50"
                onClick={onClose}
            />
            <div className="z-20 bg-white rounded-lg p-4 max-w-[30rem]">
                <div className="flex justify-between items-center mb-4">
                    <h2 className="text-4xl font-bold">Favorites</h2>
                    <button
                        className="text-xl font-bold ring-2 ring-black rounded-full w-8 h-8 flex items-center justify-center"
                        onClick={onClose}
                        aria-label="Close">
                        X
                    </button>
                </div>

                <div className="grid sm:grid-cols-1 gap-4 bg-gray-300 min-h-[40rem] min-w-[24rem] max-h-[calc(100vh_-_16rem)]  overflow-y-auto">
                    {favorites.length === 0 ? (
                        <div className="text-center text-2xl font-bold self-center">
                            No items found
                        </div>
                    ) : null}
                    {favorites.map((item, index) => (
                        <Product key={index} item={item} selected={true} />
                    ))}
                </div>
            </div>
        </div>
    );
};

export default FavoritesModal;
