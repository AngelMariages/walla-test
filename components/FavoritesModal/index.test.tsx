import FavoritesModal from '.';
import { act, getByTestId, getByText, render } from '@testing-library/react';

describe('FavoritesModal', () => {
    it('should render correctly', () => {
        const { container } = render(
            <FavoritesModal visible={true} onClose={jest.fn()} favorites={[]} />
        );
        expect(container).toMatchSnapshot();
    });

    it('should call onClose if close btn clicked', async () => {
        const closeFn = jest.fn();
        const { container } = render(
            <FavoritesModal visible={true} onClose={closeFn} favorites={[]} />
        );
        expect(container).toMatchSnapshot();
        expect(container).toMatchSnapshot();

        expect(container.querySelector('h2')?.innerHTML).toBe('Favorites');
        const closeModal = getByTestId(container, 'close-fav-button');

        expect(closeFn).not.toHaveBeenCalled();

        await act(() => {
            closeModal.click();
        });

        expect(closeFn).toBeCalledTimes(1);
    });

    it('should not render if visible is false', () => {
        const { container } = render(
            <FavoritesModal
                visible={false}
                onClose={jest.fn()}
                favorites={[]}
            />
        );
        expect(container.innerHTML).toBe('');
    });

    it('should render favorites passed', () => {
        const favorites = [
            {
                title: 'Test',
                price: 100,
                image: 'https://test.com',
                description: 'Test description',
                email: 'test@test.com',
            },
        ];
        const { container } = render(
            <FavoritesModal
                visible={true}
                onClose={jest.fn()}
                favorites={favorites}
            />
        );
        expect(container).toMatchSnapshot();

        expect(container.querySelector('h2')?.innerHTML).toBe('Favorites');



        expect(getByText(container, 'Test')).toBeTruthy();
        expect(getByText(container, '100 â‚¬')).toBeTruthy();
        expect(getByText(container, 'Test description')).toBeTruthy();
    });
});
