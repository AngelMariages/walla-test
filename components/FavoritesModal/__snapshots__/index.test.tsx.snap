// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesModal should call onClose if close btn clicked 1`] = `
<div>
  <div
    class="z-10 fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center"
  >
    <div
      class="fixed inset-0 bg-black bg-opacity-50"
    />
    <div
      class="z-20 bg-white rounded-lg p-4 max-w-[24rem] md:max-w-[30rem]"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <h2
          class="text-4xl font-bold"
        >
          Favorites
        </h2>
        <button
          aria-label="Close"
          class="text-xl font-bold ring-2 ring-black rounded-full w-8 h-8 flex items-center justify-center"
          data-testid="close-fav-button"
        >
          X
        </button>
      </div>
      <div
        class="pt-4 pb-4 border"
      >
        <form
          data-testid="search-form"
        >
          <div
            class="ml-4 md:ml-8 flex gap-2 md:gap-4"
          >
            <input
              class="rounded-md border border-gray-300 px-2 py-1"
              data-testid="search-input"
              placeholder="Search"
              type="text"
            />
            <button
              class="ml-2"
              type="submit"
            >
              <svg-url
                alt="search"
              />
            </button>
          </div>
        </form>
      </div>
      <div
        class="grid sm:grid-cols-1 gap-4 bg-gray-300 min-h-[28rem] min-w-[20rem] md:min-w-[24rem] max-h-[calc(100vh_-_16rem)] overflow-y-auto"
        data-testid="favorites-modal-list"
      >
        <div
          class="text-center text-2xl font-bold self-center"
        >
          No items found
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesModal should call onClose if close btn clicked 2`] = `
<div>
  <div
    class="z-10 fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center"
  >
    <div
      class="fixed inset-0 bg-black bg-opacity-50"
    />
    <div
      class="z-20 bg-white rounded-lg p-4 max-w-[24rem] md:max-w-[30rem]"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <h2
          class="text-4xl font-bold"
        >
          Favorites
        </h2>
        <button
          aria-label="Close"
          class="text-xl font-bold ring-2 ring-black rounded-full w-8 h-8 flex items-center justify-center"
          data-testid="close-fav-button"
        >
          X
        </button>
      </div>
      <div
        class="pt-4 pb-4 border"
      >
        <form
          data-testid="search-form"
        >
          <div
            class="ml-4 md:ml-8 flex gap-2 md:gap-4"
          >
            <input
              class="rounded-md border border-gray-300 px-2 py-1"
              data-testid="search-input"
              placeholder="Search"
              type="text"
            />
            <button
              class="ml-2"
              type="submit"
            >
              <svg-url
                alt="search"
              />
            </button>
          </div>
        </form>
      </div>
      <div
        class="grid sm:grid-cols-1 gap-4 bg-gray-300 min-h-[28rem] min-w-[20rem] md:min-w-[24rem] max-h-[calc(100vh_-_16rem)] overflow-y-auto"
        data-testid="favorites-modal-list"
      >
        <div
          class="text-center text-2xl font-bold self-center"
        >
          No items found
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesModal should render correctly 1`] = `
<div>
  <div
    class="z-10 fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center"
  >
    <div
      class="fixed inset-0 bg-black bg-opacity-50"
    />
    <div
      class="z-20 bg-white rounded-lg p-4 max-w-[24rem] md:max-w-[30rem]"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <h2
          class="text-4xl font-bold"
        >
          Favorites
        </h2>
        <button
          aria-label="Close"
          class="text-xl font-bold ring-2 ring-black rounded-full w-8 h-8 flex items-center justify-center"
          data-testid="close-fav-button"
        >
          X
        </button>
      </div>
      <div
        class="pt-4 pb-4 border"
      >
        <form
          data-testid="search-form"
        >
          <div
            class="ml-4 md:ml-8 flex gap-2 md:gap-4"
          >
            <input
              class="rounded-md border border-gray-300 px-2 py-1"
              data-testid="search-input"
              placeholder="Search"
              type="text"
            />
            <button
              class="ml-2"
              type="submit"
            >
              <svg-url
                alt="search"
              />
            </button>
          </div>
        </form>
      </div>
      <div
        class="grid sm:grid-cols-1 gap-4 bg-gray-300 min-h-[28rem] min-w-[20rem] md:min-w-[24rem] max-h-[calc(100vh_-_16rem)] overflow-y-auto"
        data-testid="favorites-modal-list"
      >
        <div
          class="text-center text-2xl font-bold self-center"
        >
          No items found
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesModal should render favorites passed 1`] = `
<div>
  <div
    class="z-10 fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center"
  >
    <div
      class="fixed inset-0 bg-black bg-opacity-50"
    />
    <div
      class="z-20 bg-white rounded-lg p-4 max-w-[24rem] md:max-w-[30rem]"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <h2
          class="text-4xl font-bold"
        >
          Favorites
        </h2>
        <button
          aria-label="Close"
          class="text-xl font-bold ring-2 ring-black rounded-full w-8 h-8 flex items-center justify-center"
          data-testid="close-fav-button"
        >
          X
        </button>
      </div>
      <div
        class="pt-4 pb-4 border"
      >
        <form
          data-testid="search-form"
        >
          <div
            class="ml-4 md:ml-8 flex gap-2 md:gap-4"
          >
            <input
              class="rounded-md border border-gray-300 px-2 py-1"
              data-testid="search-input"
              placeholder="Search"
              type="text"
            />
            <button
              class="ml-2"
              type="submit"
            >
              <svg-url
                alt="search"
              />
            </button>
          </div>
        </form>
      </div>
      <div
        class="grid sm:grid-cols-1 gap-4 bg-gray-300 min-h-[28rem] min-w-[20rem] md:min-w-[24rem] max-h-[calc(100vh_-_16rem)] overflow-y-auto"
        data-testid="favorites-modal-list"
      >
        <div
          class="m-5 p-5 border bg-slate-50 rounded-2xl shadow-2xl"
        >
          <div
            class="w-full h-80 mb-4 relative overflow-hidden rounded-2xl"
          >
            <img
              alt="Test"
              class="md:hover:scale-110 transition-transform duration-500 ease-out"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Ftest.com&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Ftest.com&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Ftest.com&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Ftest.com&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Ftest.com&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Ftest.com&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Ftest.com&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Ftest.com&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Ftest.com&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
            <svg-url
              classname="bg-white p-2 rounded-full absolute top-4 right-4 w-10 duration-500 ease-in-out transition-colors fill-red-500"
              data-testid="favicon"
            />
          </div>
          <p
            class="text-lg font-bold"
          >
            Test
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
